buildbot/master/files/config/factories.py

```diff
@@ -109,6 +113,17 @@ def make_step(self, command):
             elif arg == './etc/ci/upload_nightly.sh':
                 step_kwargs['logEnviron'] = False
                 step_env += envs.upload_nightly
+                if self.is_windows:
+                    # s3cmd on Windows only works within msys
+                    step_env['MSYSTEM'] = 'MSYS'
```

Comment:
This is a stylistic change we can always make later, so I'm OK with this as-is if we need the PR for windows nightlies now.